app/actions/remote/channel.ts

```diff
@@ -49,6 +51,98 @@ export type MyChannelsRequest = {
     error?: unknown;
 }
 
+export type ChannelMembersRequest = {
+    members?: ChannelMembership[];
+    error?: unknown;
+}
+
+export async function removeMemberFromChannel(serverUrl: string, channelId: string, userId: string) {
+    try {
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const client = NetworkManager.getClient(serverUrl);
+
+        await client.removeFromChannel(userId, channelId);
+        await deleteChannelMembership(operator, userId, channelId);
+
+        return {error: undefined};
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientErrorProps);
+        return {error};
+    }
+}
+
+export async function fetchChannelMembersByIds(serverUrl: string, channelId: string, userIds: string[], fetchOnly = false): Promise<ChannelMembersRequest> {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const members = await client.getChannelMembersByIds(channelId, userIds);
+
+        if (!fetchOnly) {
+            const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+            if (operator && members.length) {
+                const modelPromises: Array<Promise<Model[]>> = [];
+                const membership = members.map((u) => ({
+                    channel_id: channelId,
+                    user_id: u.user_id,
+                    scheme_admin: u.scheme_admin,
+                }));
+                modelPromises.push(operator.handleChannelMembership({
+                    channelMemberships: membership,
+                    prepareRecordsOnly: true,
+                }));
+
+                const models = await Promise.all(modelPromises);
+                await operator.batchRecords(models.flat());
```

Comment:
You would need these lines if you manipulated different models.  Here, you have just the ChannelMembership model. Thus, you can  set `prepareRecordsOnly` to `false`.

See [app/actions/remote/preference.ts line 30](https://github.com/mattermost/mattermost-mobile/blob/7aa5bd0611e89e2d20b8bbfdfc589cd20de94ede/app/actions/remote/preference.ts#L36) for example